src/components/CalendarContainer.tsx

```diff
@@ -191,7 +191,11 @@ function _CalendarContainer<T extends ICalendarEventBase>({
       if ((direction === 'LEFT' && !theme.isRTL) || (direction === 'RIGHT' && theme.isRTL)) {
         setTargetDate(targetDate.add(modeToNum(mode, targetDate), 'day'))
       } else {
-        setTargetDate(targetDate.add(modeToNum(mode, targetDate) * -1, 'day'))
+        if (mode === 'month') {
+          setTargetDate(targetDate.add(targetDate.date() * -1, 'day'))
+        } else {
+          setTargetDate(targetDate.add(modeToNum(mode, targetDate) * -1, 'day'))
+        }
```

Comment:
Can you add this logic to `modeToNum` function, not here?

Reply:
I wanted to but the problem is that `modeToNum` doesn't know if it's a swipe forward or back and this condition is only for swipe back.